pyiron_base/generic/hdfio.py

```diff
@@ -27,6 +28,19 @@
 __status__ = "production"
 __date__ = "Sep 1, 2017"
 
+def _is_ragged_array(value: Union[np.ndarray, list]) -> bool:
+    """
+    Checks whether array or list of lists is ragged.
+
+    Args:
+        value (ndarray/list): array to check
+
+    Returns:
+        bool: True if elements of value are not all of the same shape
+    """
+    shape_lst = [np.shape(sub) for sub in value]
+    return len(set(shape_lst)) > 1
```

Comment:
I was curious what was faster: `set`, `numpy.unique`, or `pandas.unique`. Operating on `[np.random.randint(50) for _ in range(2000)]` it was `set` *by an order of magnitude!*

Anyhow, no change here, I just thought this was interesting.